src/util/cout_message.cpp

```diff
@@ -144,49 +144,51 @@ void gcc_message_handlert::print(
   const std::string &message,
   const source_locationt &location)
 {
-  const irep_idt file=location.get_file();
-  const irep_idt line=location.get_line();
-  const irep_idt column=location.get_column();
-  const irep_idt function=location.get_function();
-
-  std::string dest;
+  message_handlert::print(level, message);
 
-  if(!function.empty())
+  if(verbosity >= level)
   {
-    if(!file.empty())
-      dest += string(messaget::bold) + id2string(file) + ":" +
-              string(messaget::reset);
-    if(dest!="")
-      dest+=' ';
-    dest += "In function " + string(messaget::bold) + '\'' +
-            id2string(function) + '\'' + string(messaget::reset) + ":\n";
-  }
+    // gcc appears to send everything to cerr
+    auto &out = std::cerr;
 
-  if(!line.empty())
-  {
-    dest += string(messaget::bold);
+    const irep_idt file = location.get_file();
+    const irep_idt line = location.get_line();
+    const irep_idt column = location.get_column();
+    const irep_idt function = location.get_function();
+
+    if(!function.empty())
+    {
+      if(!file.empty())
+        out << string(messaget::bold) << id2string(file) << ':'
```

Comment:
The use of `string(...)` should no longer be necessary, and neither is `id2string(...)`

Reply:
The id2string is removed.
Unfortunately, something like the string() is needed; the problem is that we are feeding into an std::ostream directly, an operator does not have access to the message handler.